homeassistant/components/dwd_weather_warnings/sensor.py

```diff
@@ -65,190 +63,110 @@ def setup_platform(hass, config, add_entities, discovery_info=None):
 
     api = DwdWeatherWarningsAPI(region_name)
 
-    sensors = [
-        DwdWeatherWarningsSensor(api, name, condition)
-        for condition in config[CONF_MONITORED_CONDITIONS]
-    ]
+    # Build sensor list and activate update only for the first one
+    sensors = []
+    call_update = True
```

Comment:
This variable doesn't make sense. Are you attempting to only have one sensor trigger a data update? That's what @Throttle does.

Reply:
Yes, that's what i want to do.
The API holds the data for both sensors and therefore only needs to be called once to get new data for both sensors.
My first attempt was to just `@Throttle` the `def update(self):` but with this i saw the debug log entry twice. After i added this additional parameter i got what i wanted to see.
Are there options for `@Throttle`?

Reply:
Got it. You can use `@Throttle` to  do what you want, but really, the better way to do this is to have a separate "data object" that retrieves data for everything at once; then, sensors can be updated from that single source. Depending on the complexity of what you're trying to do, the `DataUpdateCoordinator` may help: https://developers.home-assistant.io/docs/integration_fetching_data#coordinated-single-api-poll-for-data-for-all-entities

Reply:
I wanted to keep this pull request as small as possible, therefore i implemented my solution.

You're right the "data object" would be the better solution. But i think it should be made as an own pull request, not mixing it with my API movement.

Reply:
This still doesn't seem correct. Use `@Throttle`.

Reply:
I'm using `@Throttle` for the `DwdWeatherWarningsSensor` `def update(self)` function (see line 182). `call_update` is my workaround to execute the `self._api.update()` (line 186) only for the first sensor, because the api stores the value for both sensors.

Reply:
The throttle should not be on the sensor entity update method but on a method that calls the api that is shared by all sensor entites. Eg make a new wrapper class for the api that just has an update method and accepts the api instance. Pass this wrapped api instance to all entities.

```py
class WrappedDvdWWAPI:

    def __init__(self, api):
        self.api = api

    @Throttle(SCAN_INTERVAL)
    def update(self):
        self.api.update()
```

Reply:
I wrapped the api as you said. I hope i got you right.